godoclint: new rule and potential breaks with next Golangci-lint release

*This is a heads up before the upcoming Golangci-lint release.*

The new version of `godoclint` (See [`v0.11.1`](https://github.com/godoc-lint/godoc-lint/releases/tag/v0.11.1)) which will be shipped with the next Golangci-lint release, [introduces a new rule named `require-stdlib-doclink`](https://github.com/godoc-lint/godoc-lint?tab=readme-ov-file#require-stdlib-doclink) to enforce using proper doc links when referencing stdlib symbols.

Since all `godoclint` rules are enabled in `.golangci.yml`, upgrading Golangci-lint will result in new issues/errors (expand for details). The fix is basically surrounding the suggested symbols with square brackets to make them doc links. Needless to say you can also disable this new rule by adding `require-stdlib-doclink` to `linters.settings.godoclint.disable`  in `.golangci.yml`, but IMHO keeping it helps with much nicer godocs.

*If you're up to for the fix, I'd be happy to contribute back to Librarian.* 🍻


### Issues 
<details><summary>Details</summary>
<p>

```
internal/sidekick/parser/discovery/discovery.go:184:1: text "json.Unmarshaler" should be replaced with "[json.Unmarshaler]" to link to stdlib type (godoclint)
// UnmarshalJSON implements the json.Unmarshaler interface.
^
internal/sidekick/parser/discovery/discovery.go:350:1: text "json.Unmarshaler" should be replaced with "[json.Unmarshaler]" to link to stdlib type (godoclint)
// UnmarshalJSON implements the json.Unmarshaler interface.
^
internal/sidekick/parser/discovery/discovery.go:368:1: text "json.Unmarshaler" should be replaced with "[json.Unmarshaler]" to link to stdlib type (godoclint)
// UnmarshalJSON implements the json.Unmarshaler interface.
^
internal/sidekick/parser/discovery/discovery.go:409:1: text "json.Unmarshaler" should be replaced with "[json.Unmarshaler]" to link to stdlib type (godoclint)
// UnmarshalJSON implements the json.Unmarshaler interface.
^
internal/sidekick/parser/discovery/discovery.go:463:1: text "json.Unmarshaler" should be replaced with "[json.Unmarshaler]" to link to stdlib type (godoclint)
// UnmarshalJSON implements the json.Unmarshaler interface.
^
internal/sidekick/parser/discovery/discovery.go:474:1: text "json.Unmarshaler" should be replaced with "[json.Unmarshaler]" to link to stdlib type (godoclint)
// UnmarshalJSON implements the json.Unmarshaler interface.
^
internal/sidekick/sidekick/command.go:196:1: text "os.Stdout" should be replaced with "[os.Stdout]" to link to stdlib variable (godoclint)
// printUsage prints the usage of the command to os.Stdout.
^
7 issues:
* godoclint: 7
```

</p>
</details> 

cc @julieqiu

--- a/internal/sidekick/parser/discovery/discovery.go
+++ b/internal/sidekick/parser/discovery/discovery.go
@@ -181,7 +181,7 @@ type scope struct {
 	Description string
 }
 
-// UnmarshalJSON implements the json.Unmarshaler interface.
+// UnmarshalJSON implements the [json.Unmarshaler] interface.
 func (a *auth) UnmarshalJSON(data []byte) error {
 	// Pull out the oauth2 scopes and turn them into nice structs.
 	// Ignore other auth information.
@@ -347,7 +347,7 @@ type property struct {
 
 type propertyList []*property
 
-// UnmarshalJSON implements the json.Unmarshaler interface.
+// UnmarshalJSON implements the [json.Unmarshaler] interface.
 func (pl *propertyList) UnmarshalJSON(data []byte) error {
 	// In the discovery doc, properties are a map. Convert to a list.
 	var m map[string]*schema
@@ -365,7 +365,7 @@ func (pl *propertyList) UnmarshalJSON(data []byte) error {
 
 type resourceList []*resource
 
-// UnmarshalJSON implements the json.Unmarshaler interface.
+// UnmarshalJSON implements the [json.Unmarshaler] interface.
 func (rl *resourceList) UnmarshalJSON(data []byte) error {
 	// In the discovery doc, resources are a map. Convert to a list.
 	var m map[string]*resource
@@ -406,7 +406,7 @@ func (r *resource) init(parentFullName string, topLevelSchemas map[string]*schem
 
 type methodList []*method
 
-// UnmarshalJSON implements the json.Unmarshaler interface.
+// UnmarshalJSON implements the [json.Unmarshaler] interface.
 func (ml *methodList) UnmarshalJSON(data []byte) error {
 	// In the discovery doc, resources are a map. Convert to a list.
 	var m map[string]*method
@@ -460,7 +460,7 @@ func (m *method) init(topLevelSchemas map[string]*schema) error {
 	return nil
 }
 
-// UnmarshalJSON implements the json.Unmarshaler interface.
+// UnmarshalJSON implements the [json.Unmarshaler] interface.
 func (m *method) UnmarshalJSON(data []byte) error {
 	type T method // avoid a recursive call to UnmarshalJSON
 	if err := json.Unmarshal(data, (*T)(m)); err != nil {
@@ -471,7 +471,7 @@ func (m *method) UnmarshalJSON(data []byte) error {
 
 type parameterList []*parameter
 
-// UnmarshalJSON implements the json.Unmarshaler interface.
+// UnmarshalJSON implements the [json.Unmarshaler] interface.
 func (pl *parameterList) UnmarshalJSON(data []byte) error {
 	// In the discovery doc, resources are a map. Convert to a list.
 	var m map[string]*parameter

--- a/internal/sidekick/sidekick/command.go
+++ b/internal/sidekick/sidekick/command.go
@@ -193,7 +193,7 @@ func newCommand(
 	return c
 }
 
-// printUsage prints the usage of the command to os.Stdout.
+// printUsage prints the usage of the command to [os.Stdout].
 func (c *command) printUsage() {
 	if len(c.longDescription) > 0 {
 		fmt.Println(c.longDescription)

